frontend/src/metabase/static-viz/components/XYChart/XYChart.tsx

```diff
@@ -79,7 +84,7 @@ export const XYChart = ({
   const margin = calculateMargin(
     yTickWidths.left,
     yTickWidths.right,
-    xTicksDimensions.height,
+    xTicksDimensions.height + style.axes.labels.fontSize * 2,
```

Comment:
`calculateMargin` function has `labelFontSize` parameter which purpose is to preserve the space for labels when they are rendered. Seems like you wanted to increase this space so it would be better to rename `labelFontSize` to something like `labelSize` and provide the correct value there

Reply:
Let me think about this a bit more what's the best way to communicate this change in the code. What I was trying to achieve here, basically is to preserve the new 24px margin between the tick labels and the x-axis labels while always showing the x-axis label (which was hidden when rotated previously) according to the new design. So I need to let the calculator know that the chart is longer to take the x-axis label which was previously hidden when rotated into account.

|               | Normal                                                                                                         | 90deg rotated                                                                                                  |
|---------------|----------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------|
| Design        | ![image](https://user-images.githubusercontent.com/1937582/189084353-e1dfe730-e56a-48a3-90fa-7672918d788a.png) | ![image](https://user-images.githubusercontent.com/1937582/189084417-bc7e7577-6591-4934-983d-ca32472cc8ba.png) |
| w/ this line  | ![image](https://user-images.githubusercontent.com/1937582/189084751-7631bbfa-18d3-46ac-8755-8eae09e4990e.png) | ![image](https://user-images.githubusercontent.com/1937582/189084824-4f5c3cf9-7378-4637-acee-78aed266bf4c.png) |
| w/o this line | ![image](https://user-images.githubusercontent.com/1937582/189085041-d0e7d43f-0cf8-4a0b-8e83-7bb302343d29.png) | ![image](https://user-images.githubusercontent.com/1937582/189084976-3e75272e-0fa8-45cb-9fb7-ba82536729d2.png) |

Reply:
Oh, now I got it, I thought it belongs to the label since it uses `labels.fontSize`.
Something like `const ticksMargin = ...` which you will add instead of `style.axes.labels.fontSize * 2` would be a bit clearer